### PR TITLE
Add Configuration#dig() and ConfigView#dig()

### PR DESCRIPTION
**Pull request by gpakosz**
Getting closer to winning your eternal gratitude 😄

### CODE REVIEW

File: lib/nanoc/base/views/config_view.rb [hunk 1]
```diff
@@ -47,5 +47,11 @@ def each(&block)
       @context.dependency_tracker.bounce(unwrap, attributes: true)
       @config.each(&block)
     end
+
+    # @see Hash#dig
+    def dig(*keys)
+      @context.dependency_tracker.bounce(unwrap, attributes: keys)
```
Comment (ddfreyne) [inline lib/nanoc/base/views/config_view.rb]: `attributes` is a list of top-level attributes that are touched, so this should be `keys.first` instead. Nanoc only tracks dependencies of keys one level deep.

(`keys` will also work, but generate unnecessary dependencies.)

File: spec/nanoc/base/views/config_view_spec.rb [hunk 2]
```diff
@@ -115,7 +115,25 @@
...
+    end
+
+    context 'with existing keys' do
+      let(:keys) { %i[foo bar] }
+      it { is_expected.to eql(:baz) }
+    end
+
+    context 'with non-existing keys' do
+      let(:keys) { %i[foo baz bar] }
+      it { is_expected.to eql(nil) }
```
Comment (ddfreyne) [inline spec/nanoc/base/views/config_view_spec.rb]: Prefer `be_nil` over `eql(nil)`

**Comment by ddfreyne** [comment]
I’ve applied the changes and merged it into master! Thanks for the quick contribution!

**Comment by gpakosz** [comment]
Thanks for the merge!

**Pull request merged**